kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -143,28 +133,13 @@ public State next() {
     }
   }
 
-  private void updateRotationTimers() {
-    long lastRotate = time.milliseconds();
-    if (log.isDebugEnabled() && rotateIntervalMs > 0) {
-      log.debug("Update last rotation timer. Next rotation for {} will be in {}ms", tp, rotateIntervalMs);
-    }
-    if (rotateScheduleIntervalMs > 0) {
-      nextScheduledRotate = DateTimeUtils.getNextTimeAdjustedByDay(lastRotate, rotateScheduleIntervalMs, timezone);
```

Comment:
I hear you. We should give users the option to schedule rotations, making clear at the same time that they abolish exactly-once this way. 

I think reasonable defaults according to the discussion above are: 

`rotate.interval.ms` to default to `partition.duration.ms` for users that don't care about setting another interval (e.g. they want everything under hourly time-buckets where files are rolled based on size below that)
`schedule.interval.ms` to be off by default (`<0`) to aim for exactly-once by default unless you change it explicitly.